ante/tx_listener.go

```diff
@@ -0,0 +1,35 @@
+package ante
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	evmtypes "github.com/cosmos/evm/x/vm/types"
+)
+
+type PendingTxListener func(common.Hash)
+
+type TxListenerDecorator struct {
+	pendingTxListener PendingTxListener
+}
+
+// NewTxListenerDecorator creates a new TxListenerDecorator with the provided PendingTxListener.
+// CONTRACT: must be put at the last of the chained decorators
+func NewTxListenerDecorator(pendingTxListener PendingTxListener) TxListenerDecorator {
```

Comment:
Why this was migration to our side? I can imagine that this comes from a lib correct? Maybe we can reuse from source by importing